File: src/main/java/dev/psygamer/wireframe/util/collection/FreezableLinkedHashMap.java
package dev.psygamer.wireframe.util.collection;

import java.util.LinkedHashMap;
import java.util.Map;

public class FreezableLinkedHashMap <K, V> extends FreezableMap<K, V> {
	
	public FreezableLinkedHashMap() {
		super(new LinkedHashMap<>());
	}
	
	public FreezableLinkedHashMap(final int initialCapacity) {
		super(new LinkedHashMap<>(initialCapacity));
	}
	
	public FreezableLinkedHashMap(final int initialCapacity, final float loadFactor) {
		super(new LinkedHashMap<>(initialCapacity, loadFactor));
	}
	
	public FreezableLinkedHashMap(final Map<? extends K, ? extends V> map) {
		super(new LinkedHashMap<>(map));
	}
	
	@Override
	public FreezableMap<K, V> copy() {
		final FreezableMap<K, V> map = new FreezableLinkedHashMap<>(this);
		
		if (isFrozen())
			map.freeze();
		
		return map;
	}
}

File: src/main/kotlin/dev/psygamer/wireframe/util/collection/FreezableLinkedHashMap.kt
package dev.psygamer.wireframe.util.collection

class FreezableLinkedHashMap<K, V> : FreezableMap<K, V> {

	constructor() : super(LinkedHashMap<K, V>())
	constructor(initialCapacity: Int) : super(LinkedHashMap<K, V>(initialCapacity))
	constructor(initialCapacity: Int, loadFactor: Float) : super(LinkedHashMap<K, V>(initialCapacity, loadFactor))
	constructor(map: Map<out K, V>) : super(LinkedHashMap<K, V>(map))
}